File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function solution(n, k) {
    let answer = 0;
    const arr = n.toString(k).split(0);

    for (const a of arr) {
        if (a === "") continue;
        if (isPrime(+a)) answer++;
    }

    return answer;
}

function isPrime(n) {
    if (n === 1) {
        return false;
    }

    for (let i = 2; i <= Math.sqrt(n); i++) {
        if (n % i === 0) {
            return false;
        }
    }

    return true;
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        String[] arr = Integer.toString(n, k).split("0");
        
        for(String s : arr) {
            if(s.isEmpty()) {
                continue;
            } else {
                long l = Long.parseLong(s);
                
                if(isPrime(l)) answer++;
            }
        }

        return answer;
    }

    public boolean isPrime(long n) {
        if(n == 1) return false;

        for(int i=2; i<=Math.sqrt(n); i++) {
            if(n%i == 0) return false;
        }

        return true;
    }
}